### ISSUE TITLE
Constraints test does not capture error in Conformance class: Application Schema, Protected Sites simple

### ISSUE THREAD

**Issue by jonherrmann** [inspire-eu-validation/ets-repository#44]
Moved from https://github.com/interactive-instruments/etf-webapp/issues/86

@sMorrone

I validated an example file containing the erroneus designation value “http://inspire.ec.europa.eu/codelist/IUCNDesignationValue/wildernessAreaFakeValue” for a  PS with  designationScheme=”http://inspire.ec.europa.eu/codelist/DesignationSchemeValue/IUCN" and  it passes validation (relevant log at http://35.157.17.37/etf-webapp/v2/TestResults/d69a2d0e-10e8-4d6c-8060-d522ace5eb51/log).
Would you please check?
Stefania

**Comment by jonherrmann** [comment]
@sMorrone
Thanks for the issue report Stefania, we will check this.

**Comment by cportele** [comment]
Hi Stefania, 

it looks like you tested the Conformance Class "GML application schemas, Protected Sites" which does not include the test case for the code list values. This is part of the Conformance Class "Application schema, Protected Sites Simple". The unknown value is identified in the test for this conformance class. See here: http://35.157.17.37/etf-webapp/reports/62aaf956-cdcb-4204-83f1-de59d7929707.

This is related to the following test case: https://github.com/inspire-eu-validation/data-ps/blob/3.2/ps-as/code-list-values.md

Could you check the behaviour with your test data using the ETS for the Conformance Class "Application schema, Protected Sites Simple"?

Please note that the next version of the software supports ordering the group of conformance classes / ETSs and the "basic tests" - which typically will not be selected directly, but only automatically as a dependency - will be at the bottom of the list. Also, you will be able to select multiple ETSs to execute during a single test run.

@sMorrone

**Comment by sMorrone** [comment]
Still passing validation
![passingvalidation](https://cloud.githubusercontent.com/assets/13329248/24203498/32a6936c-0f16-11e7-858c-381aee51cfac.png)

**Comment by cportele** [comment]
Yes, but with manual checks. The test probably has a message like:

> The property 'designation' has a value 'wildernessAreaFakeValue' that is not one of the pre-defined values in the INSPIRE data specification. The same value is used by x other features in the dataset, too. Extensions to the pre-defined values are allowed, but must not overlap with one of the pre-defined values and be consistent with the specification of the property. Please review the definition of the value.

Since the extensibility of designationValue is open, any additional value is allowed as long as its semantics does not overlap with one of the pre-defined values. So, it would be an error to report a failure, it requires a manual check.

**Comment by jonherrmann** [comment]
Just to be sure @cportele : the exception in the log 
```
2017-03-22T14:37:35 INFO "bxerr:BASX0000 - null Nested exception: null -  - 1308 - 36 - Stopped at ., 1308/36:&#xA;[bxerr:BASX0000] null Nested exception: null"
```
can be classified as 'issue 41 geometry parse' error or is this a new kind of error class?

**Comment by sMorrone** [comment]
Maybe it is not clear the nature of the issue. I'm trying to be clearer posting the excerpt of the file which should cause a failure (ERROR in my view - not simple warning with manual test). That because the designationscheme is IUCN and relevant  http://inspire.ec.europa.eu/codelist/IUCNDesignationValue code list is not extensible).

<ps:siteDesignation>
                <ps:DesignationType>
                    <ps:designationScheme
                        xlink:href="http://inspire.ec.europa.eu/codelist/DesignationSchemeValue/IUCN"
                        xlink:title="IUCN"/>
                    <ps:designation
                        xlink:href="http://inspire.ec.europa.eu/codelist/IUCNDesignationValue/wildernessAreaFakeValue"
                        xlink:title="wildernessArea"/>
                </ps:DesignationType>
            </ps:siteDesignation>

The warning message I get from the validator is " The property 'designation' has a value 'http://inspire.ec.europa.eu/codelist/IUCNDesignationValue/wildernessAreaFakeValue' that is not one of the pre-defined values in the INSPIRE data specification. The same value is used by 0 other features in the dataset, too. Extensions to the pre-defined values are allowed, but must not overlap with one of the pre-defined values and be consistent with the specification of the property. Please review the definition of the value."
According to A1.6 test I should receive an error

Stefania

**Comment by cportele** [comment]
@jonherrmann - no this is a different issue, looks like there is an issue in the GmlGeoX, deegree or JTS libraries. I did not notice this and have not seen it before. @sMorrone - for testing that exception, could we have access to the file that you used for testing?

@sMorrone - I see. In my test (I had been using the v3 schema) that issue has been detected: "XML document 'ps-cl-error-0.xml', ProtectedSite 'xxx': The constraint 'Sites must use designations from an appropriate designation scheme.' is violated. For designation scheme 'IUCN' the following disallowed value(s) have been used: wildernessAreaFakeValue. Allowed values are: habitatSpeciesManagementArea, managedResourceProtectedArea, nationalPark, naturalMonument, ProtectedLandscapeOrSeascape, strictNatureReserve, wildernessArea." So there must be an issue in the test for the v4 schemas that you have been using. We will check. Thanks for reporting.

**Comment by sMorrone** [comment]
here it comes
[PSWithCodeListError.zip](https://github.com/inspire-eu-validation/ets-repository/files/861975/PSWithCodeListError.zip)

**Comment by cportele** [comment]
@sMorrone - thanks. I found the error. I now get an error message: http://35.157.17.37/etf-webapp/reports/2717562e-88fb-4177-b859-6a50a9a0acad?lang=en. Does this look better?

@jonherrmann - regarding the exception, can you take a look in the next days. Locally I get a proper TR.invalidGeometry message: "Invalid polygon. Outer ring of polygon is clockwise within element Polygon, (gml:id: m.PS.EMA120001 with coordinates: LINESTRING (239013.326300 7018756.671600,239012.481400 70187...)".

**Comment by sMorrone** [comment]
Perfect 👍

**Comment by jonherrmann** [comment]
Sure, i will open a new issue for it.

**Comment by jonherrmann** [comment]
ii internal note:
Machine dev6 is using an outdated version of GmlGeoX (ci build 14). 
Testing the file on dev7 reports the "Invalid polygon..." error. See Report EID32c7fb57-3072-4b38-b4d8-48e8a91fc7bf.
Issue is already fixed in ETF 2.0.0-beta.
